skbio/stats/ordination/_principal_coordinate_analysis.py

```diff
@@ -152,6 +152,14 @@ def pcoa(
                 "distance matrix is large.",
                 RuntimeWarning,
             )
+        elif method == "eigh" and distmat.data.shape[0] > 10:
```

Comment:
I am not sure if this warning is necessary, or if the triggering threshold (n > 10) is too small. It is quite common that a distance matrix with hundreds of samples is fed to PCoA. That won't cause long computation time, will it?

Reply:
There are several `matrix_data.shape[0] > 10` lines in the added code. I thought that it means the input distance matrix has more than 10 samples, instead of the user specified more than 10 dimensions. Is this understanding correct?

Reply:
This specific test is within `if dimensions == 0` block.
Which has the implict meaning of  "convert dimensions to size of DM".
So testing on either would be equivalent (if we tested after the assignment; the test actually happens before dimensions is changed).

Reply:
@sfiligoi depends on the use-case. If we are going to be using eigh as a proxy for embedding search, then yes absolutely -- the dimensionality dictates distance approximation accuracy. 

But we don't currently support this type of functionality yet

Reply:
It's just a warning in case a user does not explicitly provide the number of dimensions it wants.
(So, nothing breaks in those cases)
Moreover, if they do provide the number of dimensions, there is no warning, ever.

That said, I have no strong opinion if we want/need a warning or not.

But would definitely prefer if we threated "eight" and "fsvd" the same, as the cost considerations are now identical.